### ISSUE TITLE
Warnings related to `esmpy` package `__author__`, `__homepage__`, and `__obsoletes__`

### ISSUE THREAD

**Issue by tomvothecoder** [esmf-org/esmf#140]
Hello, I am receiving `esmpy` warnings (via `xesmf`) in my package's test suite. Just wanted to make sure the dev team is aware of this.

```python
../../miniconda3/envs/xcdat_ci/lib/python3.10/site-packages/esmpy/__init__.py:106
  /home/vo13/miniconda3/envs/xcdat_ci/lib/python3.10/site-packages/esmpy/__init__.py:106: DeprecationWarning: Implicit None on return values is deprecated and will raise KeyErrors.
    __author__ = msg["Author"]

../../miniconda3/envs/xcdat_ci/lib/python3.10/site-packages/esmpy/__init__.py:107
  /home/vo13/miniconda3/envs/xcdat_ci/lib/python3.10/site-packages/esmpy/__init__.py:107: DeprecationWarning: Implicit None on return values is deprecated and will raise KeyErrors.
    __homepage__ = msg["Home-page"]

../../miniconda3/envs/xcdat_ci/lib/python3.10/site-packages/esmpy/__init__.py:108
  /home/vo13/miniconda3/envs/xcdat_ci/lib/python3.10/site-packages/esmpy/__init__.py:108: DeprecationWarning: Implicit None on return values is deprecated and will raise KeyErrors.
    __obsoletes__ = msg["obsoletes"]

```

**Comment by billsacks** [comment]
@tomvothecoder thank you for reporting this. I am having trouble reproducing this, probably due to my lack of experience enabling deprecation warnings in python.

Things I have tried (from https://docs.python.org/3/library/warnings.html and elsewhere), using python 3.11:
- Importing esmpy with python started normally
- Importing esmpy with python started with `python -Wd`
- Importing esmpy with python started with `python -W "default::DeprecationWarning"`
- Importing esmpy with python started with `python -Walways`

None of these led to the deprecation warnings you noted above. Would you be able to describe how to reproduce these warnings?

**Comment by tomvothecoder** [comment]
I tried your commands and also can't reproduce the issue. I only notice it in my repo's [test suite](https://github.com/xCDAT/xcdat/actions/runs/5326491519/jobs/9648600797#step:9:26).

Here is extra info that might be helpful.

### Conda environment:
```bash
esmf                      8.4.2           mpi_mpich_h742b71d_100    conda-forge
esmpy                     8.4.2              pyhc1e730c_1    conda-forge
...
python                    3.11.4          hab00c5b_0_cpython    conda-forge
...
xesmf                     0.7.1              pyhd8ed1ab_0    conda-forge
```

### Warnings
```bash
../../miniconda3/envs/xcdat_dev/lib/python3.11/site-packages/esmpy/__init__.py:106
  /home/vo13/miniconda3/envs/xcdat_dev/lib/python3.11/site-packages/esmpy/__init__.py:106: DeprecationWarning: Implicit None on return values is deprecated and will raise KeyErrors.
    __author__ = msg["Author"]

../../miniconda3/envs/xcdat_dev/lib/python3.11/site-packages/esmpy/__init__.py:107
  /home/vo13/miniconda3/envs/xcdat_dev/lib/python3.11/site-packages/esmpy/__init__.py:107: DeprecationWarning: Implicit None on return values is deprecated and will raise KeyErrors.
    __homepage__ = msg["Home-page"]

../../miniconda3/envs/xcdat_dev/lib/python3.11/site-packages/esmpy/__init__.py:108
  /home/vo13/miniconda3/envs/xcdat_dev/lib/python3.11/site-packages/esmpy/__init__.py:108: DeprecationWarning: Implicit None on return values is deprecated and will raise KeyErrors.
    __obsoletes__ = msg["obsoletes"]
```

### Related code lines:
https://github.com/esmf-org/esmf/blob/7babfadf548e183c4ecece177444a1350fe73cc1/src/addon/esmpy/src/esmpy/__init__.py#L105-L108

### What I think is causing the issue
"Author", "Home-page" and "obsoletes" aren't set in `pyproject.toml` but is being referenced in `esmpy.__init__.py`.

https://github.com/esmf-org/esmf/blob/7babfadf548e183c4ecece177444a1350fe73cc1/src/addon/esmpy/pyproject.toml#L5-L23

### Minimum example:
```python
import importlib.metadata as ilm

msg = ilm.metadata("esmpy")

print(msg.__dict__)
# {'policy': Compat32(), '_headers': [('Metadata-Version', '2.1'), ('Name', 'esmpy'), ('Version', '8.4.2'), ('Summary', 'ESMF Python interface'), ('Maintainer-email', 'ESMF Core Team <esmf_support@ucar.edu>'), ('License', 'University of Illinois-NCSA'), ('Requires-Python', '>=3.7'), ('License-File', 'LICENSE'), ('Requires-Dist', 'numpy'), ('Requires-Dist', 'importlib-metadata ; python_version < "3.8"'), ('Provides-Extra', 'testing'), ('Requires-Dist', "pytest ; extra == 'testing'"), ('Requires-Dist', "pytest-json-report ; extra == 'testing'")], '_unixfrom': None, '_payload': '', '_charset': None, 'preamble': None, 'epilogue': None, 'defects': [], '_default_type': 'text/plain'}


msg["Author"]  # Returns None
msg["Home-page"]  # Returns None
msg["obsoletes"]  # Returns None

```

### Related issues from other repos
  - https://github.com/pypa/twine/issues/977
  - https://github.com/python/cpython/issues/103624

**Comment by billsacks** [comment]
Thank you for your very detailed information, @tomvothecoder ! As far as I can tell, these three attributes aren't needed for anything, so I have simply removed them. Please let me know if you run into any additional, related issues.

We're about to have our 8.5.0 release freeze; I'm not sure if the fix will make it into the 8.5.0 release or will need to wait for 8.6.0.

**Comment by valeriupredoi** [comment]
@billsacks cheers very much! Has the fix made it to 8.5.0 or 8.6.0? Many thanks :beer:

**Comment by billsacks** [comment]
This fix is in 8.5.0 and all following versions.

**Comment by valeriupredoi** [comment]
> This fix is in 8.5.0 and all following versions.

brilliant, many thanks @billsacks :beer:
